server/executor/trace_poller.go

```diff
@@ -190,7 +196,13 @@ func (tp tracePoller) runAssertions(job PollingRequest) {
 func (tp tracePoller) handleTraceDBError(job PollingRequest, err error) (bool, string) {
 	run := job.run
 
-	pp := *tp.ppGetter.GetDefault(job.ctx).Periodic
+	profile := tp.ppGetter.GetDefault(job.ctx)
```

Comment:
sometimes the context can get cancelled in the middle of this func call, causing a nil pointer panic. This new approach handles the case more gracefully